### ISSUE TITLE
javafx class integration in JDK17 upgrade recipe

### ISSUE THREAD

**Issue by Kushank24** [openrewrite/rewrite-migrate-java#472]
## What problem are you trying to solve?
I am running the JDK17 update recipe. But once the recipe is run and I am trying to test it out I am getting the following error: 
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project topology: Compilation failure: Compilation failure:
[ERROR] package javafx.util does not exist
[ERROR] package javafx.util does not exist
[ERROR] cannot find symbol
[ERROR] symbol: class Pair
[ERROR] location: class 

## Describe the solution you'd like
The solution should be incorporated in the JDK17 upgrade recipe only so that the solution automatically gets remdified.

## Have you considered any alternatives or workarounds?
Tried using java.util.Map.Entry but also some inn that sense i needed to change the user code

**Comment by timtebeek** [comment]
Thanks for logging an issue! As discussed on Slack there's likely a couple changes we'd need to make for JavaFX projects adopting Java 17; would you mind breaking down those changes a bit in this issue? Things like adding a dependency, changing packages or classes, or switching to a different runtime. That way we're better able to start codifying such changes as recipes.

**Comment by timtebeek** [comment]
I think adding the required dependencies is the most straightforward way we can keep supporting JavaFX if that's used.

If folks only use one or two classes (such as Pair here), then it might make sense to use `ChangeType` instead to an alternative.

**Comment by Kushank24** [comment]
Yeah correct Tim. I resolved the issue by adding the following dependencies in the parent pom:
`<dependency>
 <groupId>org.openjfx</groupId>
 <artifactId>javafx-base</artifactId>
 <version>17.0.7</version>
</dependency>
<dependency>
 <groupId>org.openjfx</groupId>
 <artifactId>javafx-controls</artifactId>
 <version>17.0.7</version>
</dependency>
<dependency>
 <groupId>org.openjfx</groupId>
 <artifactId>javafx-fxml</artifactId>
 <version>17.0.7</version>
</dependency>`

And it worked.

**Comment by timtebeek** [comment]
Looks like those could be added with a declarative recipe if needed. That would then close out this issue for folks similarly migrating. Is that something you'd want to contribute?
